sys/mips/mips/locore.S

```diff
@@ -163,9 +163,18 @@ VECTOR(_locore, unknown)
 	/*
 	 * Enable userspace cycle-count, cycle-count resolution, and CPU
 	 * number registers available via rdhwr.
+	 * Also enable beri specific statcounters.
 	 */
 	MFC0	t0, MIPS_COP_0_HWRENA
-	or	t0, MIPS_HWRENA_CC | MIPS_HWRENA_CCRES | MIPS_HWRENA_CPUNUM | (7 << 4)
+	or	t0, MIPS_HWRENA_CC | MIPS_HWRENA_CCRES | MIPS_HWRENA_CPUNUM | (7 << 4) \
+	| MIPS_BERI_HWRENA_RESET_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_ICACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_DCACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_L2CACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_MEM_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_TAGCACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_L2MASTER_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_TAGMASTER_STATCOUNTERS
```

Comment:
As far as I understand, `MIPS_HWRENA_CC`, `MIPS_HWRENA_CCRES` and `MIPS_HWRENA_CPUNUM` are standard MIPS-isms, required to be present in all MIPS implementation, and disabled by default. It looks like the write to CP0 Register 7, Select 0 that is necessary to enable them is only done for beri kernels, but I don't know that they aren't turned on in other cases, maybe not using an easily grep-able macro... The statcounters I added are beri specific. Currently the hardware has them enabled by default to keep libstatcounters happy,  but I intend to change that after this gets folded into the beri/cheri kernels.